sycl/unittests/program_manager/BuildLog.cpp

```diff
@@ -94,11 +94,16 @@ TEST(BuildLog, OutputNothingOnLevel1) {
     GTEST_SKIP_("Test is not supported on this platform");
   }
 
+  const sycl::device Dev = Plt.get_devices()[0];
+  if (!Dev.has(sycl::aspect::online_compiler)) {
+    GTEST_SKIP_("Test is not supported on this device due to missing support "
+                "for sycl::aspect::online_compiler");
```

Comment:
I thought the unit tests had no real device support and everything was mocked. Are we mocking devices with and without `aspect::online_compiler` ?   Or perhaps I'm misunderstanding.    And, since the latter is most likely, does that also mean that `get_devices()` will never return an empty vector? Does our mocking guarantee at least one?